apps/src/templates/Certificate.jsx

```diff
@@ -23,152 +23,142 @@ const blankCertificates = {
   mee_timecraft: require('@cdo/static/MC_Hour_Of_Code_Certificate_mee_timecraft.png')
 };
 
-class Certificate extends Component {
-  constructor() {
-    super();
-    this.state = {
-      personalized: false
-    };
-  }
-
-  static propTypes = {
-    tutorial: PropTypes.string,
-    certificateId: PropTypes.string,
-    randomDonorTwitter: PropTypes.string,
-    responsiveSize: PropTypes.oneOf(['lg', 'md', 'sm', 'xs']).isRequired,
-    under13: PropTypes.bool,
-    children: PropTypes.node
-  };
+function Certificate(props) {
+  const [personalized, setPersonalized] = useState(false);
+  const nameInputRef = useRef(null);
 
-  isMinecraft = () =>
+  const isMinecraft = () =>
     /mc|minecraft|hero|aquatic|mee|mee_empathy|mee_timecraft/.test(
-      this.props.tutorial
+      props.tutorial
     );
-  isAIOceans = () => /oceans/.test(this.props.tutorial);
+  const isAIOceans = () => /oceans/.test(props.tutorial);
 
-  personalizeCertificate(session) {
+  const personalizeCertificate = session => {
     $.ajax({
       url: '/v2/certificate',
       type: 'post',
       dataType: 'json',
       data: {
         session_s: session,
-        name_s: this.nameInput.value
+        name_s: nameInputRef.current.value
       }
     }).done(response => {
       if (response.certificate_sent) {
-        this.setState({personalized: true});
+        setPersonalized(true);
       }
     });
-  }
+  };
 
-  render() {
-    const {
-      responsiveSize,
-      tutorial,
-      certificateId,
-      randomDonorTwitter,
-      under13,
-      children
-    } = this.props;
-    const certificate = certificateId || 'blank';
-    const personalizedCertificate = `${
-      dashboard.CODE_ORG_URL
-    }/api/hour/certificate/${certificate}.jpg`;
-    const blankCertificate =
-      blankCertificates[tutorial] || blankCertificates.hourOfCode;
-    const imgSrc = this.state.personalized
-      ? personalizedCertificate
-      : blankCertificate;
-    const certificateLink = `https:${
-      dashboard.CODE_ORG_URL
-    }/certificates/${certificate}`;
-    const desktop =
-      responsiveSize === ResponsiveSize.lg ||
-      responsiveSize === ResponsiveSize.md;
-    const headingStyle = desktop ? styles.heading : styles.mobileHeading;
-    const certificateStyle = desktop ? styles.desktopHalf : styles.mobileFull;
+  const {
+    responsiveSize,
+    tutorial,
+    certificateId,
+    randomDonorTwitter,
+    under13,
+    children
+  } = props;
 
-    const facebook = queryString.stringify({
-      u: certificateLink
-    });
+  const certificate = certificateId || 'blank';
```

Comment:
style nit: Could make `blank` the `defaultProp` for `certificateId` and avoid this, but it's old behavior